R/fabricate.R

```diff
@@ -121,12 +152,36 @@ fabricate_data_single_level <- function(data = NULL,
 
   if (!is.null(N)) {
     if (length(N) != 1) {
-      stop(
-        "At the top level, ",
-        ID_label,
-        ", you must provide a single number to N."
-      )
+      if(is.null(ID_label)) {
+        stop(
+          "At the top level, you must provide a single number to N."
```

Comment:
Since this is just an error message, you could use a ifelse to make this a little more legible.